controller/array_action/array_mediator_ds8k.py

```diff
@@ -435,3 +565,16 @@ def validate_supported_capabilities(self, capabilities):
                 capabilities)
 
         logger.debug("Finished validating capabilities.")
+
+    def _generate_snapshot_response(self, api_snapshot, source_volume_name):
+        return Snapshot(capacity_bytes=int(api_snapshot.cap),
+                        snapshot_id=self._generate_volume_scsi_identifier(api_snapshot.id),
+                        snapshot_name=api_snapshot.name,
+                        array_address=self.service_address,
+                        volume_name=source_volume_name,
+                        is_ready=True,
+                        array_type=self.array_type)
+
+    def validate_flashcopy(self, flashcopy_id):
```

Comment:
consider renaming to something like `is_valid_flashcopy` (if you think it's better)